common/quotas/global/algorithm/requestweighted.go

```diff
@@ -538,3 +578,18 @@ func weighted[T numeric](newer, older T, weight float64) T {
 type numeric interface {
 	constraints.Integer | constraints.Float
 }
+
+// non-sync version of sync.Once, for easier unlocking
```

Comment:
I'd make it explicit in the comments there "Do NOT use this for concurrent cases"

Reply:
that is part of why I didn't make it a public type, yea.  not intended for reuse.